### PR TITLE
[Gtk Pipeline] Made New Template Dialog use Header Bar if possible

### PR DESCRIPTION
**Pull request by cra0zy**
Just a bit of Linux integration when possible :D, depending on HeaderBar value it will load one of two things:

withheaderbar
withoutheaderbar

### CODE REVIEW

**Comment by cra0zy** [comment]
@tomspilman @KonajuGames When you get the chance to look at my PRs, start with this one please

**Comment by tomspilman** [comment]
What is the "HeaderBar value"?  Is that some accessibility feature?

**Comment by cra0zy** [comment]
In code, just a bool, in Gtk, it's integration with titlebar.

**Comment by tomspilman** [comment]
Is HeaderBar a common UI pattern in Linux?  Just never seen any UI where the action buttons are at the top of the window.

**Comment by cra0zy** [comment]
https://blogs.gnome.org/mclasen/2014/07/28/a-talk-in-9-images/

**Comment by cra0zy** [comment]
In Gnome 3 a lot stuff are up, look at the middle of all the images in the link above.

File: Tools/Pipeline/Gtk/Global.cs [hunk 1]
```diff
@@ -32,10 +34,22 @@ public static void Initalize()
                 DesktopEnvironment = line;
             }
             #endif
+
+            UseHeaderBar = Global.GtkMajorVersion >= 3 && Global.GtkMinorVersion >= 12 && Global.DesktopEnvironment == "GNOME";
```
Comment (cra0zy) [inline Tools/Pipeline/Gtk/Global.cs]: @tomspilman With tis line I check if headerbar shoul bne used. It checks if Gtk is 3.12 or higher and if it;s running GNOME as desktop enviorment.

**Comment by tomspilman** [comment]
So this is some feature of the GNOME desktop.  As long as it doesn't affect Mac and is consistent with other apps on Linux... no problem.

**Comment by cra0zy** [comment]
I can garantee it will not affect anything but GNOME

**Comment by cra0zy** [comment]
Side note, gtk is maintained by GNOME, https://github.com/GNOME/gtk

**Comment by tomspilman** [comment]
Ok... merging!

**Pull request merged**